lib/li_box4d.c

```diff
@@ -122,10 +122,21 @@ li_box4d_affine(
 * Apply a projective pinhole transformation to a box4d.
 */
 LIBOX4 *
-li_box4d_projective_pinhole(LIBOX4 ibox, double focal, double px, double py)
+li_box4d_projective_pinhole(LIBOX4 ibox, double focal, double px, double py, char inverse)
 {
-	LIMAT44 pmat;
+	LIMAT44 mat;
+	double *pmat;
 
-	li_matrix_44_set_from_pinhole_projection(pmat, px, py, focal, focal, 0);
+	li_matrix_44_set_from_pinhole_projection(mat, px, py, focal, focal, 0);
+	if ( inverse )
+	{
+		LIMAT44 imat;
+		li_matrix_44_inverse(imat, mat);
+		pmat = imat;
+	}
+	else
+	{
+		pmat = mat;
+	}
```

Comment:
I would more efficiently do :
```
if ( inverse )
{
	li_matrix_44_set_from_pinhole_unprojection(mat, px, py, focal, focal, 0);
}
else
{
	li_matrix_44_set_from_pinhole_projection(mat, px, py, focal, focal, 0);
}
```
Otherwise, LGTM !

Reply:
Good call. Changed.